src/igraph/constructors/from_file.py

```diff
@@ -0,0 +1,228 @@
+"""
+IGraph library.
+"""
+
+
+__license__ = """
+Copyright (C) 2006-2012  Tamás Nepusz <ntamas@gmail.com>
+Pázmány Péter sétány 1/a, 1117 Budapest, Hungary
+
+Copyright (C) 2021- igraph development team
+
+This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program; if not, write to the Free Software
+Foundation, Inc.,  51 Franklin Street, Fifth Floor, Boston, MA
+02110-1301 USA
+"""
+import gzip
+from shutil import copyfileobj
+
+from igraph.utils import (
+    named_temporary_file,
+)
+
+
+def construct_graph_from_adjacency_file(
+    cls, f, sep=None, comment_char="#", attribute=None, *args, **kwds
+):
+    """Constructs a graph based on an adjacency matrix from the given file.
+
+    Additional positional and keyword arguments not mentioned here are
+    passed intact to L{Adjacency}.
```

Comment:
`L{Adjacency}` is a broken reference now that we stripped this function out of the `Graph` class.